modules/node_modules/@ncigdc/components/AddToCartButtonAll.js

```diff
@@ -0,0 +1,101 @@
+// @flow
+
+import React from 'react';
+import { connect } from 'react-redux';
+
+import { parseFilterParam } from '@ncigdc/utils/uri';
+import LocationSubscriber from '@ncigdc/components/LocationSubscriber';
+import ShoppingCartIcon from 'react-icons/lib/fa/shopping-cart';
+import DownCaretIcon from 'react-icons/lib/fa/caret-down';
+import TrashIcon from 'react-icons/lib/fa/trash';
+
+import { addAllFilesInCart, fetchFiles, removeAllInCart, toggleFilesInCart } from '@ncigdc/dux/cart';
+
+import Row from '@ncigdc/uikit/Flex/Row';
+import Dropdown from '@ncigdc/uikit/Dropdown';
+import Button from '@ncigdc/uikit/Button';
+
+import styled from '@ncigdc/theme/styled';
+
+const DropDownCaret = styled.span({
+  backgroundColor: '#FFF',
+  borderTopRightRadius: '4px',
+  borderBottomRightRadius: '4px',
+});
+
+const DropDownStyle = {
+  position: 'absolute',
+  right: 'initial',
+  left: '-25px',
+  marginTop: '5px',
+  borderRadius: '4px',
+};
+
+const DropDownItemStyle = {
+  padding: '3px 20px',
+  margin: '5px 0',
+  fontWeight: 'normal',
+};
+
+const CartButton = styled(Button, ({
+  padding: '0 5px',
+  color: '#FFF',
+  borderRadius: '4px',
+  borderTopRightRadius: '0px',
+  borderBottomRightRadius: '0px',
+  cursor: 'pointer',
+  backgroundColor: ({ active }) => (active ? '#255425' : '#0d95a1'),
+  ':hover': {
+    backgroundColor: ({ active }) => (active ? '#255425' : '#0d95a1'),
+  },
+}));
+
+const filesInCart = (edges, files) => {
+  if (files.length < edges.length) return false;
+  return edges.every((edge) => files.some((file) => file.file_id === edge.file_id));
+};
+
+const AddToCartButtonAll = ({ edges, files, total, dispatch }) => (
+  <LocationSubscriber>{ctx => {
+    const { filters, ...query } = ctx.query || {};
+    const currentFilters = parseFilterParam(filters, { content: [] });
+
+    return (
+      <Row>
+        <CartButton
+          active={filesInCart(edges, files)}
+          onClick={() => (filesInCart(edges, files)
```

Comment:
`edges` should probably be renamed to something else

It's like naming an array `array`, right?